java/src/org/openqa/selenium/grid/distributor/selector/DefaultSlotSelector.java

```diff
@@ -53,6 +53,11 @@ public Set<SlotId> selectSlot(
                 .thenComparingDouble(NodeStatus::getLoad)
                 // Then last session created (oldest first), so natural ordering again
                 .thenComparingLong(NodeStatus::getLastSessionCreated)
+                // Then sort by stereotype browserVersion (descending order). SemVer comparison with
+                // considering empty value at first.
+                .thenComparing(
+                    Comparator.comparing(
+                        NodeStatus::getBrowserVersion, NodeStatus::semVerComparator))
```

Comment:
The `semVerComparator` returns `true` or `false`. This is more suitable for a filter. Which is done below, through the matcher.

Reply:
This is still needed since multiple matches returned from SlotMatcher should be sorted to move the latest version to top